common/txmgr/types/tx.go

```diff
@@ -32,3 +37,75 @@ const (
 type Transaction interface {
 	GetID() string
 }
+
+type NewTx[ADDR types.Hashable, TX_HASH types.Hashable] struct {
```

Comment:
Could we take this opportunity to rename this struct?
NewTx is a very confusing name. This is the type that is sent by the user/caller.
Could we call it TxRequest?
Client calls txmgr as CreateEthTransaction(txRequest, ...)

Seems like a more cleaner name to me.

First, do you agree?
If yes, I'm ok if this happens in this PR, or a follow-up one.

Reply:
I don't feel strongly about the naming - so I'm happy to change it - let me make a ticket so we can pick it up in a separate PR (i'd like to keep changes limited in this PR)
https://smartcontract-it.atlassian.net/browse/BCI-1209